### ISSUE TITLE
causal = True

### ISSUE THREAD

**Issue by wajihullahbaig** [lucidrains/linear-attention-transformer#5]
Naive question!
 causal = True , is this used to create a mask that trims/clips the top right values of the attention matrix?

**Comment by lucidrains** [comment]
@wajihullahbaig yup! in linear attention, you do this with a cumulative sum instead of the triangular mask!

**Comment by wajihullahbaig** [comment]
> @wajihullahbaig yup! in linear attention, you do this with a cumulative sum instead of the triangular mask!

Much appreciated for the reply!

Thanks!
